lib/formatters/BaseFormatter.js

```diff
@@ -0,0 +1,138 @@
+const logTags = require('../logTags');
+const chalk = require('chalk');
+const colors = require('../appenders/colors');
+
+const outRes = logTags.CATEGORY.OUTBOUND_RESPONSE;
+const inReq = logTags.CATEGORY.INBOUND_REQUEST;
+const outReq = logTags.CATEGORY.OUTBOUND_REQUEST;
+const inRes = logTags.CATEGORY.INBOUND_RESPONSE;
+const inReqH = logTags.CATEGORY.INBOUND_REQUEST_HEALTH;
+const outResH = logTags.CATEGORY.OUTBOUND_RESPONSE_HEALTH;
+const unexpectedErr = logTags.CATEGORY.UNEXPECTED_ERROR;
+
+class BaseFormatter {
+  constructor(color = false, appendDate = false, requestType = '') {
+    this.color = color;
+    this.appendDate = appendDate;
+    this.requestType = requestType;
+  }
+
+  formatObject(obj) {
+    throw new Error('Not implemented');
+  }
+
+  getPrefix(obj) {
+    const prefixMapper = {
+      [inReq.TAG]: obj => {
+        return this.getInRequestPrefix(obj);
+      },
+      [outReq.TAG]: obj => {
+        return this.getOutRequestPrefix(obj);
+      },
+      [inReqH.TAG]: obj => {
+        return this.getInRequestHealthPrefix(obj);
+      },
+      [inRes.TAG]: obj => {
+        return this.getInResponsePrefix(obj);
+      },
+      [outRes.TAG]: obj => {
+        return this.getOutResponsePrefix(obj);
+      },
+      [outResH.TAG]: obj => {
+        return this.getOutResponseHealthPrefix(obj);
+      },
+      [unexpectedErr.TAG]: obj => {
+        return this.getUnexpectedErrorPrefix(obj);
+      }
+    };
+
+    return prefixMapper[this.requestType] ? prefixMapper[this.requestType](obj) : '';
+  }
+
+  getInRequestPrefix(obj) {
+    let prefix = `[${obj.requestId}] ${inReq.MSG} ${obj.method} ${obj.path}`;
+    if (this.appendDate) {
+      prefix = `[${new Date().toISOString()}] ${prefix}`;
+    }
+    if (this.color) {
+      prefix = chalk[[colors[inReq.TAG]]](prefix);
+    }
+
+    return `${prefix}`;
```

Comment:
Is prefix something else than string here?

Reply:
Fixed